### DISCUSSION TITLE
Configuration Options for Treesitter <-> Color/Style Mappings

### DISCUSSION THREAD

**Discussion by jtyler76** [EdenEast/nightfox.nvim#388] [Q&A] [answered]
Hi there, I've been exploring various customization options with Nightfox, and it's going fairly well, but I'm stumped on how one might modify the Treesitter mappings found here (https://github.com/EdenEast/nightfox.nvim/blob/main/lua/nightfox/group/modules/treesitter.lua) from the user configuration options.

My use case is that I would like `@function.builtin` and `@keyword.return` to be represented by different colours (and maybe styles).

As far as I've been able to work out, there is no user-facing method of accomplishing this. Both map to `fg = syn.builtin0`, which can be changed through configuration options, but that doesn't allow you to have one blue and one red let's say.

Any guidance would be greatly appreciated! I've been using Nightfox for a few years now and it's by far my favourite option for colours in Neovim!

**Comment by EdenEast** [comment, marked as answer]
Not sure I fully understand your question but if I understand correctly you want some way to set `@function.builtin` and `@keyword.return` differently. A colorscheme's spec also contains the underlying palette. You can set this like:

```lua
require("nightfox").setup({
  groups = {
    all = {
      ["@function.builtin"] = { fg = "palette.blue" },
      ["@keyword.return"] = { fg = "palette.red" },
    },
  },
})
```

**Reply by jtyler76** [reply]
Thanks, setting some of the `@...` items explicitly like this helped to resolve the issues I was having!
